woocommerce/woocommerce
# Move CES data store to @woocommerce/customer-effort-score

### All Submissions:






### Changes proposed in this Pull Request:



This PR moves the CES data store currently found in `woocommerce/admin/client` to the `@woocommerce/customer-effort-score` package. This builds on the work started in #37112.

**No functionality should be impacted by this change... this is just a package file refactoring.**

_Note: There are non-generic bits of functionality in the CES data store (controlling whether specific modals are shown, etc.)... these should be abstracted out in the future to a pattern where `@woocommerce/customer-effort-score` contains no feature-specific details. For now though, we can live with this._

Closes #37247.



### How to test the changes in this Pull Request:




Note: make sure tracking is enabled under WooCommerce > Settings > Advanced > WooCommerce.com

In order for CES snackbars and such to show up, you need to have not already dismissed/completed the CES for a particular feature. To reset things, you can delete the `woocommerce_ces_shown_for_actions` transient (use the WCA Test Helper).

1. Load this branch and run `pnpm install && pnpm run build` and make sure everything builds correctly.
2. Smoke test WCAdmin pages and make sure that any CES-related functionality works correctly.
   
    * Import CES snackbar - Go to Products > All Products; click "Import"; without continuing, click back to Products > All Products (see testing instructions on #36656)
    * Filtering snackbar - Use advanced filter in Analytics > Variations; enter advanced filter criteria and click "Filter"
    * CES footer on new product editor page - make sure "Share feedback" button brings up feedback modal.

3. Open your console and check if the following exist under `window.wc.customerEffortScore` (there will be more, these are the new ones added in this PR):

- `STORE_KEY`
- `QUEUE_OPTION_NAME`



### Other information:




### FOR PR REVIEWER ONLY:
